modules/dnn/src/tensorflow/tf_importer.cpp

```diff
@@ -318,21 +334,35 @@ void setKSize(LayerParams &layerParams, const tensorflow::NodeDef &layer)
     if (hasLayerAttr(layer, "ksize"))
     {
         const tensorflow::AttrValue& val = getLayerAttr(layer, "ksize");
-        int dimX, dimY, dimC;
+        int dimX, dimY, dimC, dimD;
         int layout = getDataLayout(layer);
         if (layout == DATA_LAYOUT_NCHW)
         {
             dimC = 1; dimY = 2; dimX = 3;
         }
+        else if (layout == DATA_LAYOUT_NDHWC)
+        {
+            dimD = 1; dimY = 2; dimX = 3; dimC = 4;
+        }
         else
         {
             dimY = 1; dimX = 2; dimC = 3;
         }
```

Comment:
I think that it's better to check that `layout == DATA_LAYOUT_NHWC` here and raise an exception if not.

Reply:
In Test_TensorFlow.read_inception layout = DATA_LAYOUT_UNKNOWN

Reply:
In Test_TensorFlow_layers.conv layout = DATA_LAYOUT_NCHW